runtime/platform/test/executor_pal_static_runtime_override_test.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <gtest/gtest.h>
+
+#include <executorch/runtime/platform/platform.h>
+#include <executorch/runtime/platform/test/pal_spy.h>
+
+namespace {
+PalSpy spy = PalSpy();
+
+void pal_init(void) {
+  spy.init();
+}
+
+et_timestamp_t pal_current_ticks(void) {
+  return spy.current_ticks();
+}
+
+et_tick_ratio_t pal_ticks_to_ns_multiplier(void) {
+  return spy.ticks_to_ns_multiplier();
+}
+
+void pal_emit_log_message(
+    et_timestamp_t timestamp,
+    et_pal_log_level_t level,
+    const char* filename,
+    ET_UNUSED const char* function,
+    size_t line,
+    const char* message,
+    ET_UNUSED size_t length) {
+  spy.emit_log_message(
+      timestamp, level, filename, function, line, message, length);
+}
+
+void* pal_allocate(size_t size) {
+  return spy.allocate(size);
+}
+
+void pal_free(void* ptr) {
+  spy.free(ptr);
+}
+
+// Statically register PAL impleementation.
+bool registration_result = executorch::runtime::register_pal(
```

Comment:
how do we know this doesn't get removed by the linker in some different build config, without a `static` qualifier?